File: ecommerce/lib/models/product.dart
import 'package:ecommerce/models/reviews.dart';

class Product {
  String? nama;
  int? harga;
  int? stock;
  String? image;
  String? imageUrl;
  String? kadaluarsa;
  int? berat;
  num? rating;
  List<Review>? reviews;

  Product({
    this.nama,
    this.harga,
    this.stock,
    this.image,
    this.imageUrl,
    this.kadaluarsa,
    this.berat,
    this.rating,
    this.reviews,
  });

  factory Product.fromJson(dynamic json) {
    return Product(
      nama: json['nama'],
      harga: json['harga'],
      stock: json['stock'],
      image: json['image'],
      imageUrl: json['imageUrl'],
      kadaluarsa: json['kadaluarsa'],
      berat: json['berat'],
      rating: json['rating'],
    );
  }
}

File: crud-laravel-10/app/Models/Product.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Product extends Model
{
    use HasFactory;

    protected $fillable = [
        'image',
        'nama',
        'harga',
        'stock',
        'jenis',
        'kadaluarsa',
        'berat',
        'rating',
    ];

    public function reviews()
    {
        return $this->hasMany(Review::class, 'product_id');
    }

    public function getImageUrlAttribute()
    {
        if (strpos($this->image, 'http') === 0) {
            return $this->image;
        }
        return asset('storage/posts/' . $this->image);
    }
}
